axelor-supplychain/src/main/resources/views/Mrp.xml

```diff
@@ -10,14 +10,14 @@
         <field name="mrpSeq"/>
         <field name="createdOn"/>
         <field name="createdBy"/>
-        <field name="stockLocation" form-view="stock-location-form" grid-view="stock-location-grid" colSpan="6"/>
-        <field name="stockLocation.company"/>
+        <field name="stockLocationFullNames" colSpan="6"/>
+        <field name="stockLocationCompanies"/>
         <field name="endDate" colSpan="6"/>
         <field name="statusSelect"/>
 	 </grid>
     
 	<form name="mrp-form" model="com.axelor.apps.supplychain.db.Mrp" title="Mrp" width="large"
-		onLoad="action-attrs-mrp-generate-all-proposals">
+		onLoad="action-attrs-mrp-generate-all-proposals" onSave="action-mrp-record-set-stocklocation-fullname-and-companies-fullname,save">
```

Comment:
stockLocationCompanies and stockLocationFullNames should be filled on stockLocationSet change, not on save.